server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -851,6 +853,10 @@ public boolean isConfigured() {
             return isSet && Objects.equals(value, getDefaultValue()) == false;
         }
 
+        public boolean isSet() {
```

Comment:
Note that I had to add this method to be able to detect the situation when the parameter `index` is set to `true` (the default) explicitly. The `isConfigured` existing method returns `true` only if the value is set and is different from the default too (which is not our case).

Reply:
`isConfigured` is not `true`...

```
public boolean isConfigured() {
            return isSet && Objects.equals(value, getDefaultValue()) == false;
        }
```

The second condition is `true` because value=true and getDefaultValue=true...so the Objects/.equals evaluates to false. This implementation of `isCOnfigured` does not really make sense to me.

Reply:
IMO the error is in `&&` used instead of `||`.

Reply:
I see, thanks for explaining this.